src/touch_save_app.c

```diff
@@ -0,0 +1,520 @@
+static void TouchSaveApp_DestroyWindow(TouchSaveAppData*);
+static void TouchSaveApp_SetupText(TouchSaveAppData*, u8 unused);
+static void TouchSaveApp_DestroyText(TouchSaveAppData*);
+static void ov30_0225D880(TouchSaveAppData *data);
+static void TouchSaveApp_SetupWaitForTextPrinter(TouchSaveAppData *data, enum TouchSaveApp_State nextState);
+static BOOL TouchSaveApp_DisplaySaveInformation(TouchSaveAppData*);
+static BOOL TouchSaveApp_AskForSave(TouchSaveAppData*);
+static BOOL TouchSaveApp_WaitForTextPrinter(TouchSaveAppData*);
+static BOOL TouchSaveApp_GetSaveConfirmation(TouchSaveAppData*);
+static BOOL TouchSaveApp_HandleSaveConfirmation(TouchSaveAppData*);
+static BOOL TouchSaveApp_PrintOverwriteMessage(TouchSaveAppData*);
+static BOOL TouchSaveApp_GetOverwriteConfirmation(TouchSaveAppData*);
+static BOOL TouchSaveApp_HandleOverwriteConfirmation(TouchSaveAppData*);
+static BOOL TouchSaveApp_PrintNotMySaveError(TouchSaveAppData*);
+static BOOL TouchSaveApp_SaveFailed(TouchSaveAppData*);
+static BOOL TouchSaveApp_PrintSavingMessage(TouchSaveAppData*);
+static BOOL TouchSaveApp_SetupWaitingIcon(TouchSaveAppData*);
+static BOOL TouchSaveApp_SaveGame(TouchSaveAppData*);
+static BOOL TouchSaveApp_PrintSavedMessage(TouchSaveAppData*);
+static BOOL TouchSaveApp_SaveSucceeded(TouchSaveAppData*);
+static BOOL TouchSaveApp_CloseApp(TouchSaveAppData*);
+static BOOL TouchSaveApp_ShouldPrintAlternateSavingMessage(TouchSaveAppData*);
+static void ov30_0225DC00(u32*, u32);
+static void ov30_0225DC08(void);
+static void ov30_0225DC18(void);
+static void ov30_0225DC28(void);
+
+static const BgTemplate ov30_0225DC64 = {
+    .x = 0,
+    .y = 0,
+    .bufferSize = 0x800,
+    .baseTile = 0,
+    .size = GF_BG_SCR_SIZE_256x256,
+    .colorMode = GX_BG_COLORMODE_16,
+    .screenBase = GX_BG_SCRBASE_0x6000,
+    .charBase = GX_BG_CHARBASE_0x04000,
+    .bgExtPltt = GX_BG_EXTPLTT_01,
+    .priority = 1,
+    .areaOver = GX_BG_AREAOVER_XLU,
+    .dummy = 0,
+    .mosaic = FALSE,
+};
+
+static const BgTemplate ov30_0225DC48 = {
+    .x = 0,
+    .y = 0,
+    .bufferSize = 0x800,
+    .baseTile = 0,
+    .size = GF_BG_SCR_SIZE_256x256,
+    .colorMode = GX_BG_COLORMODE_16,
+    .screenBase = GX_BG_SCRBASE_0x7000,
+    .charBase = GX_BG_CHARBASE_0x00000,
+    .bgExtPltt = GX_BG_EXTPLTT_01,
+    .priority = 0,
+    .areaOver = GX_BG_AREAOVER_XLU,
+    .dummy = 0,
+    .mosaic = FALSE,
+};
+
+static const BgTemplate ov30_0225DC2C = {
+    .x = 0,
+    .y = 0,
+    .bufferSize = 0x800,
+    .baseTile = 0,
+    .size = GF_BG_SCR_SIZE_256x256,
+    .colorMode = GX_BG_COLORMODE_16,
+    .screenBase = GX_BG_SCRBASE_0x6800,
+    .charBase = GX_BG_CHARBASE_0x00000,
+    .bgExtPltt = GX_BG_EXTPLTT_01,
+    .priority = 2,
+    .areaOver = GX_BG_AREAOVER_XLU,
+    .dummy = 0,
+    .mosaic = FALSE,
+};
+
+typedef BOOL (*TouchSaveAppStateFunc)(TouchSaveAppData*);
+static /*const*/ TouchSaveAppStateFunc sTouchSaveAppStateFuncs[TOUCHSAVEAPP_STATE_COUNT] = {
```

Comment:
is this comment from decomping/should this array be const?

Reply:
It could have been const since the array is never modified, but it was not const in the final build.

Reply:
does it match with const?

Reply:
No.

Reply:
then remove comment